tests/unit/common_util_tests.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+# Copyright (c) 2017 IBM. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+_common_util_tests_
+
+Unit tests for the _common_util helper functions.
+
+"""
+
+import mock
+import requests
+import unittest
+
+from cloudant import _common_util as util
+
+
+class CommonUtilTests(unittest.TestCase):
+
+    def test_get_docs(self):
+        m_sess = requests.Session()
+        m_resp = requests.Response()
+        m_resp.status_code = 200
+
+        m_get = mock.Mock(return_value=m_resp)
+        m_sess.get = m_get
+
+        url = 'http://127.0.0.1/my_db'
+        resp = util.get_docs(m_sess, url)
+
+        m_get.assert_called_once_with(url, headers=None, params={})
+
+        self.assertEquals(200, resp.status_code)
+
+    def test_get_docs_with_keys(self):
+        m_sess = requests.Session()
+        m_resp = requests.Response()
+        m_resp.status_code = 200
+
+        m_post = mock.Mock(return_value=m_resp)
+        m_sess.post = m_post
+
+        keys = ['doc1', 'doc2', 'doc3', 'doc4', 'doc5']
+        url = 'http://127.0.0.1/my_db'
+        resp = util.get_docs(m_sess, url, keys=keys)
+
+        m_post.assert_called_once_with(
+            url,
+            data='{"keys": ["doc1", "doc2", "doc3", "doc4", "doc5"]}',
+            headers={'Content-Type': 'application/json'},
+            params={}
+        )
+
+        self.assertEquals(200, resp.status_code)
+
+    def test_get_docs_with_empty_keys(self):
+        m_sess = requests.Session()
+        m_resp = requests.Response()
+        m_resp.status_code = 200
+
+        m_post = mock.Mock(return_value=m_resp)
+        m_sess.post = m_post
+
+        keys = []  # empty keys list
+        url = 'http://127.0.0.1/my_db'
+        resp = util.get_docs(m_sess, url, keys=keys)
+
+        m_post.assert_called_once_with(
+            url,
+            data='{"keys": []}',
+            headers={'Content-Type': 'application/json'},
+            params={}
+        )
+
+        self.assertEquals(200, resp.status_code)
```

Comment:
righto, I'll add another all_docs test in database_tests using an empty key list and remove these.